evennia/utils/utils.py

```diff
@@ -1021,44 +1021,34 @@ def uses_database(name="sqlite3"):
 
 
 def delay(timedelay, callback, *args, **kwargs):
-    """
-    Delay the calling of a callback (function).
+    """Delay the calling of a callback (function).
+
+    The task handler (`evennia.scripts.taskhandler.TaskHandler`) will
+    be called for persistent or non-persistent tasks.
+
+    If persistent is set to True.
+        The callback, its arguments and other keyword arguments will be saved
+        (serialized) in the database, assuming they can be. The callback will
+        be executed even after a server restart/reload, taking into account the
+        specified delay (and server down time).
+        Keep in mind that persistent tasks arguments and callback should not
+        use memory references.
+        All task's whose time delays have passed will be called on server startup.
```

Comment:
This sounds like it belongs partly in the `persistent` arg description and in a `Notes` section.